diff/util/chunk.go

```diff
@@ -0,0 +1,291 @@
+	"github.com/pingcap/tidb-tools/pkg/db"
+	"github.com/pingcap/tidb/model"
+)
+
+// chunkRange represents chunk range
+type chunkRange struct {
+	begin interface{}
+	end   interface{}
+	// for example:
+	// containB and containE is true, means [begin, end]
+	// containB is true, containE is false, means [begin, end)
+	containB bool
+	containE bool
+}
+
+// DumpJob is the struct for job of dump
+type DumpJob struct {
+	DbName string
+	Table  string
+	Column *model.ColumnInfo
+	Where  string
+	Chunk  chunkRange
+}
+
+// newChunkRange return a range struct
+func newChunkRange(begin, end interface{}, containB, containE bool) chunkRange {
+	return chunkRange{
+		begin:    begin,
+		end:      end,
+		containE: containE,
+		containB: containB,
+	}
+}
+
+func getChunksForTable(db *sql.DB, dbName, tableName string, column *model.ColumnInfo, where string, chunkSize, sample int) ([]chunkRange, error) {
+	if column == nil {
+		log.Warnf("no suitable index found for %s.%s", dbName, tableName)
+		return nil, nil
+	}
+
+	field := column.Name.O
+
+	// fetch min, max
+	query := fmt.Sprintf("SELECT %s MIN(`%s`) as MIN, MAX(`%s`) as MAX FROM `%s`.`%s` where %s",
+		"/*!40001 SQL_NO_CACHE */", field, field, dbName, tableName, where)
+
+	// get the chunk count
+	cnt, err := pkgdb.GetRowCount(db, dbName, tableName, where)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if cnt == 0 {
+		log.Infof("no data found in %s.%s", dbName, tableName)
+		return nil, nil
+	}
+
+	chunkCnt := cnt / int64(chunkSize)
+	if sample != 100 {
+		// use sampling check, can check more fragmented by split to more chunk
+		chunkCnt *= 10
+	}
+
+	var chunk chunkRange
+	if pkgdb.IsNumberType(column.Tp) {
+		var min, max sql.NullInt64
+		err := db.QueryRow(query).Scan(&min, &max)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if !min.Valid {
+			// min is NULL, means that no table data.
+			return nil, nil
+		}
+		chunk = newChunkRange(min.Int64, max.Int64+1, true, false)
+	} else if pkgdb.IsFloatType(column.Tp) {
+		var min, max sql.NullFloat64
+		err := db.QueryRow(query).Scan(&min, &max)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if !min.Valid {
+			// min is NULL, means that no table data.
+			return nil, nil
+		}
+		chunk = newChunkRange(min.Float64-0.1, max.Float64+0.1, false, false)
+	} else {
+		var min, max sql.NullString
+		err := db.QueryRow(query).Scan(&min, &max)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if !min.Valid || !max.Valid {
+			return nil, nil
+		}
+		chunk = newChunkRange(min.String, max.String, true, true)
+	}
+	return splitRange(db, &chunk, chunkCnt, dbName, tableName, column, where)
+}
+
+func splitRange(db *sql.DB, chunk *chunkRange, count int64, dbName string, table string, column *model.ColumnInfo, limitRange string) ([]chunkRange, error) {
+	var chunks []chunkRange
+
+	if count <= 1 {
+		chunks = append(chunks, *chunk)
+		return chunks, nil
+	}
+
+	if reflect.TypeOf(chunk.begin).String() == "int64" {
```

Comment:
How about `reflect.TypeOf(chunk.begin).Kind() == reflect.Int64`?

Reply:
if don't have min, the max must be null